File: back/service/src/main/java/org/webtree/mydata/service/UserService.java
package org.webtree.mydata.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.webtree.mydata.domain.User;
import org.webtree.mydata.repository.UserRepository;
import reactor.core.publisher.Mono;

@Service
public class UserService {
    private UserRepository userRepository;

    @Autowired
    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public Mono findById(String id) {
        return userRepository.findById(id);
    }
    public final Mono addUser(User user) {
        return userRepository.save(user);
    }
}

File: api/src/main/kotlin/org/webtree/mydata/api/service/UserService.kt
package org.webtree.mydata.api.service

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.webtree.mydata.api.domain.User
import org.webtree.mydata.api.repository.UserRepository
import reactor.core.publisher.Mono

@Service
class UserService(@Autowired val userRepository: UserRepository) {
    fun findById(id: String): Mono<User> {
        return userRepository.findById(id)
    }

    fun addUser(user: User): Mono<User> {
        return userRepository.save(user)
    }
}